### DISCUSSION TITLE
completion like vim ctrl-p and ctrl-n in normal text file

### DISCUSSION THREAD

**Discussion by dlintw** [helix-editor/helix#9016] [Q&A] [answered]
I often use the quick completion function in VIM, that's ctrl-p, ctrl-n which in insert mode of even normal text file.  Any tip to hack into rust code for implement such function?

**Comment by kirawi** [comment, marked as answer]
https://github.com/estin/simple-completion-language-server

You can also check old PRs:
https://github.com/helix-editor/helix/pull/4816
https://github.com/helix-editor/helix/pull/3328

**Comment by dlintw** [comment]
Thanks, I use the github and it works. But, that will let my config much longer. Whenever I found something can not work, require add a 3 lines setup to add the language to config.toml.
